src/axolotl/utils/logging.py

```
@@ -0,0 +1,54 @@
+        )
+
+    def log(self, level, msg, *args, **kwargs):
+        main_process_only = kwargs.pop("main_process_only", True)
+        kwargs.setdefault("stacklevel", 2)
+
+        if self.isEnabledFor(level):
+            if self._should_log(main_process_only):
+                msg, kwargs = self.process(msg, kwargs)
+                self.logger.log(level, msg, *args, **kwargs)
+
+    @functools.lru_cache(None)
```

Ruff (0.11.9)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)